contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,527 @@
+   */
+  function hashOfKey(
+    uint256[2] memory publicKey
+  )
+    public
+    pure
+    returns (
+      bytes32
+    )
+  {
+    return keccak256(abi.encodePacked(publicKey));
+  }
+
+  function setConfig(
+    uint16 minimumRequestBlockConfirmations,
+    uint16 maxConsumersPerSubscription,
+    uint32 stalenessSeconds,
+    uint32 gasAfterPaymentCalculation,
+    int256 fallbackLinkPrice,
+    uint96 minimumSubscriptionBalance
+  )
+    external
+    onlyOwner()
+  {
+    s_config = Config({
+      minimumRequestBlockConfirmations: minimumRequestBlockConfirmations,
+      maxConsumersPerSubscription: maxConsumersPerSubscription,
+      stalenessSeconds: stalenessSeconds,
+      gasAfterPaymentCalculation: gasAfterPaymentCalculation,
+      minimumSubscriptionBalance: minimumSubscriptionBalance
+    });
+    s_fallbackLinkPrice = fallbackLinkPrice;
+    emit ConfigSet(minimumRequestBlockConfirmations,
+      maxConsumersPerSubscription,
+      stalenessSeconds,
+      gasAfterPaymentCalculation,
+      fallbackLinkPrice,
+      minimumSubscriptionBalance
+    );
+  }
+
+  /**
+   * @notice read the current configuration of the coordinator.
+   */
+  function getConfig()
+    external
+    view
+    returns (
+      uint16 minimumRequestBlockConfirmations,
+      uint16 maxConsumersPerSubscription,
+      uint32 stalenessSeconds,
+      uint32 gasAfterPaymentCalculation,
+      int256 fallbackLinkPrice
+    )
+  {
+    Config memory config = s_config;
+    return (
+      config.minimumRequestBlockConfirmations,
+      config.maxConsumersPerSubscription,
+      config.stalenessSeconds,
+      config.gasAfterPaymentCalculation,
+      s_fallbackLinkPrice
+    );
+  }
+
+  function requestRandomWords(
+    bytes32 keyHash,  // Corresponds to a particular offchain job which uses that key for the proofs
+    uint64  subId,
+    uint64  minimumRequestConfirmations,
+    uint64  callbackGasLimit,
+    uint64  numWords,  // Desired number of random words
+    uint16  consumerID // Index into consumers to avoid SLOADing all the consumers
+  )
+    external
+    returns (
+      uint256 requestId
+    )
+  {
+    // Input validation using the subscription storage.
+    if (s_subscriptions[subId].owner == address(0)) {
+      revert InvalidSubscription();
+    }
+    // We use this consumer index to ensure that the cost remains constant no matter
+    // the number of consumers.
+    if (s_subscriptions[subId].consumers[consumerID] != msg.sender) {
+      revert InvalidConsumer(msg.sender);
+    }
+    // Input validation using the config storage word.
+    if (minimumRequestConfirmations < s_config.minimumRequestBlockConfirmations) {
+      revert RequestBlockConfsTooLow(minimumRequestConfirmations, s_config.minimumRequestBlockConfirmations);
+    }
+    if (s_subscriptions[subId].balance < s_config.minimumSubscriptionBalance) {
+      revert InsufficientBalance();
+    }
+    if (s_serviceAgreements[keyHash] == address(0)) {
```

Comment:
We could consider removing this check to save 2100 gas, the v1 contract doesnt have this check (just lets it fail silently), presumably for gas savings. At the same time we've seen a handful of cases in practice where people pass in the testnet keyhash or a docs keyhash